From 6ca200f570caf3f97075367d29fe02e7b12a3603 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:22:50 -0400
Subject: [pci] Clear NumVFs when disabling SR-IOV in sriov_init()

Message-id: <20140719042250.19438.9842.stgit@gir.stowe>
Patchwork-id: 85963
O-Subject: [RHEL7.1 PATCH 01/81] PCI: Clear NumVFs when disabling SR-IOV in sriov_init()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 045cc22e53a1a0caaa9a0b6b64246d2690049906

commit 045cc22e53a1a0caaa9a0b6b64246d2690049906
Author: ethan.zhao <ethan.kernel@gmail.com>
Date:   Wed Nov 6 22:49:13 2013 +0800

    PCI: Clear NumVFs when disabling SR-IOV in sriov_init()

    When SR-IOV is disabled (VF Enable is cleared), NumVFs is not very useful,
    so this patch clears it out to prevent confusing lspci output like that
    below.  We already clear NumVFs in sriov_disable(), and this does the same
    when we disable SR-IOV as part of parsing the SR-IOV capability.

      $ lspci -vvv -s 13:00.0
      13:00.0 Ethernet controller: Intel Corporation 82599EB 10-Gigabit
SFI/SFP+ Network Connection (rev 01)
          Capabilities: [160 v1] Single Root I/O Virtualization (SR-IOV)
              IOVCtl: Enable- Migration- Interrupt- MSE- ARIHierarchy+
              Initial VFs: 64, Total VFs: 64, Number of VFs: 64, ...

    [bhelgaas: changelog]
    Signed-off-by: ethan.zhao <ethan.kernel@gmail.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/iov.c b/drivers/pci/iov.c
index e2fbb67..9dce7c5 100644
--- a/drivers/pci/iov.c
+++ b/drivers/pci/iov.c
@@ -442,6 +442,7 @@ static int sriov_init(struct pci_dev *dev, int pos)
 
 found:
  pci_write_config_word(dev, pos + PCI_SRIOV_CTRL, ctrl);
+ pci_write_config_word(dev, pos + PCI_SRIOV_NUM_VF, 0);
  pci_read_config_word(dev, pos + PCI_SRIOV_VF_OFFSET, &offset);
  pci_read_config_word(dev, pos + PCI_SRIOV_VF_STRIDE, &stride);
  if (!offset || (total > 1 && !stride))
-- 
1.7.1